sdk/core/Azure.Core/src/Pipeline/HttpWebRequestTransport.cs

```diff
@@ -110,6 +111,11 @@ private async ValueTask ProcessInternal(HttpMessage message, bool async)
         private HttpWebRequest CreateRequest(Request messageRequest)
         {
             var request = WebRequest.CreateHttp(messageRequest.Uri.ToUri());
+
+            // Timeouts are handled by the pipeline
+            request.Timeout = Timeout.Infinite;
+            request.ReadWriteTimeout = Timeout.Infinite;
```

Comment:
https://docs.microsoft.com/en-us/dotnet/api/system.net.httpwebrequest.readwritetimeout?view=net-5.0#remarks this talks about timeouts on `Stream.Write` as well. I recall we apply NetworkTimeout on `Stream.Read`. How is Stream.Write secured by `NetworkTimeout`?

Reply:
Right now the only `Write` we do is during content sending. The content is written during the `Transport.SendAsync` call and is guarded by this timeout:

https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/Pipeline/Internal/ResponseBodyPolicy.cs#L43-L60

Reply:
Thank you for the thorough review.